src/Symfony/Component/Form/Extension/Core/Type/TimezoneType.php

```diff
@@ -38,19 +38,22 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setDefaults(array(
             'choice_loader' => function (Options $options) {
                 $regions = $options['regions'];
+                $country = $options['country'];
 
-                return new CallbackChoiceLoader(function () use ($regions) {
-                    return self::getTimezones($regions);
+                return new CallbackChoiceLoader(function () use ($regions, $country) {
+                    return self::getTimezones($regions, $country);
                 });
             },
             'choice_translation_domain' => false,
             'input' => 'string',
-            'regions' => \DateTimeZone::ALL,
+            'regions' => isset($options['country']) ? \DateTimeZone::PER_COUNTRY : \DateTimeZone::ALL,
```

Comment:
$options doesn't exist in this context, this is always evaluated to \DateTimeZone::ALL

Reply:
Right, this option needs a lazy default to evaluate the `country` value. ([See example](https://github.com/symfony/symfony/blob/b3fc3b569003ea9454449391a01d516d0a560f9a/src/Symfony/Component/Form/Extension/Core/Type/FileType.php#L99-L108))